dask/dataframe/io/parquet/fastparquet.py

```diff
@@ -344,23 +347,58 @@ def read_metadata(
         # Create `parts`
         # This is a list of row-group-descriptor dicts, or file-paths
         # if we have a list of files and gather_statistics=False
+        base_path = (base_path or "") + fs.sep
```

Comment:
If we're doing direct path manipulations, we should maybe start a HTTP server and test against it